mlflow/dspy/autolog.py

```diff
@@ -0,0 +1,79 @@
+from mlflow.dspy.save import FLAVOR_NAME
+from mlflow.tracing.provider import trace_disabled
+from mlflow.utils.annotations import experimental
+from mlflow.utils.autologging_utils import autologging_integration, safe_patch
+
+
+@experimental
+def autolog(
+    log_traces: bool = True,
+    disable: bool = False,
+    silent: bool = False,
+):
+    """
+    Enables (or disables) and configures autologging from DSPy to MLflow. Currently, MLflow
+    only supports autologging for tracing.
```

Comment:
nit: to make it more clear that it's just for the dspy flavor, not for mlflow overall

```suggestion
    Enables (or disables) and configures autologging from DSPy to MLflow. Currently, the DSPy
    flavor only supports autologging for tracing.
```